src/Streamly/Internal/Data/Stream/Concurrent/Channel/Append.hs

```diff
@@ -228,26 +230,19 @@ workLoopLIFOLimited qref sv winfo = run
 --
 -- we can even run the already queued items but they will have to be sorted in
 -- layers in the heap. We can use a list of heaps for that.
+{-# ANN enqueueAhead "HLint: ignore" #-}
 {-# INLINE enqueueAhead #-}
 enqueueAhead ::
        Channel m a
     -> IORef ([K.Stream m a], Int)
     -> (RunInIO m, K.Stream m a)
     -> IO ()
 enqueueAhead sv q m = do
-    atomicModifyIORefCAS_ q $ \ case
-        ([], n) -> ([snd m], n + 1)  -- increment sequence
-        _ -> error "enqueueAhead: queue is not empty"
-    ringDoorBell (doorBellOnWorkQ sv) (outputDoorBell sv)
-
--- enqueue without incrementing the sequence number
-{-# INLINE reEnqueueAhead #-}
-reEnqueueAhead ::
-    Channel m a -> IORef ([K.Stream m a], Int) -> K.Stream m a -> IO ()
-reEnqueueAhead sv q m = do
-    atomicModifyIORefCAS_ q $ \ case
-        ([], n) -> ([m], n)  -- DO NOT increment sequence
-        _ -> error "reEnqueueAhead: queue is not empty"
+    -- XXX The queue is LIFO. When parConcatIterate queues more than one items
+    -- to the queue it will perform a DFS style traversal. For BFS we will have
+    -- to use a FIFO data structure here. That would require another Config
+    -- option.
+    atomicModifyIORefCAS_ q $ \(xs, n) -> (snd m:xs, n)
```

Comment:
I do not understand this. Can you explain this change?

Reply:
This is `reEnqueueAhead` now?

Reply:
1. Now we can have more then one item in the queue, because iterate appends to the queue.
2. Because of that we cannot increment the sequence number on enqueue because sequence number was the number to be used for the enqueued item, now we have more than one. Instead we increment the sequence number on dequeue now.